api/v1beta2/network_types.go

```diff
@@ -757,3 +863,16 @@ func (i *IngressRule) Equals(o *IngressRule) bool {
 
 	return true
 }
+
+// ZoneType defines listener AWS Availability Zone type.
+type ZoneType string
+
+// String returns the string representation for the zone type.
+func (z ZoneType) String() string {
+	return string(z)
+}
+
+// Equal compares two zone types.
+func (z ZoneType) Equal(other ZoneType) bool {
+	return z == other
+}
```

Comment:
Is this needed?

Reply:
Not for now. FUP https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4882#discussion_r1566355658 : 
I am removing from this PR, it will introduce in WL when we'll need more complex check w/ more edge types. Thanks